microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/jaxrs/JaxRsUtils.java

```diff
@@ -133,12 +134,17 @@ public static CodeLens createURLCodeLens(String baseURL, String rootPath, String
 	}
 
 	private static CodeLens createURLCodeLens(IMethod method, IJDTUtils utils) throws JavaModelException {
-		ISourceRange r = method.getNameRange();
-		if (r == null) {
+		IAnnotation[] annotations = method.getAnnotations();
+		if (annotations == null) {
 			return null;
 		}
+		ISourceRange r = annotations[annotations.length - 1].getSourceRange();
+
 		CodeLens lens = new CodeLens();
-		final Range range = utils.toRange(method.getOpenable(), r.getOffset(), r.getLength());
+		Range range = utils.toRange(method.getOpenable(), r.getOffset(), r.getLength());

```

Comment:
I really like this approach, although it is a little buggy when the annotation spans multiple lines:
![MultiLineAnnotationJaxCodeLens](https://user-images.githubusercontent.com/22376627/202203651-c095bcf5-b972-42c6-b7da-b4038963791b.png)

(this is `VillainResource` from the completed Quarkus super heros)

Reply:
Ugh, and `IMemberValuePair` doesn't have a source range. It might be possible to do from the AST, but that complicates things.

Reply:
What is the location of the end annotation? If it's after the member value pairs, then maybe you could set the range to be from the end of the annotation to the end annotation (same spot)?

Reply:
Nice that works!! Thank you for the suggestion!